### ISSUE TITLE
[Feature request] "Find in files" option to exclude binary files and/or respect .gitignore files

### ISSUE THREAD

**Issue by leomeyer** [notepad-plus-plus/notepad-plus-plus#12342]
### Description of the Issue
"Find in files" currently only supports including/excluding file and folder names by specifying filters.

For large source trees with a multitude of text files with different extensions plus possibly large binary files this is impractical.
There could be an option "Exclude binary files" that automatically excludes certain extensions by platform (e. g. *.exe, *.dll, *.obj) and/or skips the currently processed file as soon as a byte is encountered indicating that the file is not a text file. This is somewhat depending on the file's encoding but probably not too difficult; any byte in range 0x00 - 0x1F excluding 0x09 (HTAB), 0x0A (LF) and 0x0D (CR) should trigger the skip (assuming that UTF-16 is not supported as an encoding here).

An advanced but very useful feature would be to automatically identify .gitignore files (or possibly other CSV ignore specifications) and use their rules to exclude folders and files from the tree. An option like this could greatly speed up a developer's workflow when working in large source trees with lots of generated binaries.

**Comment by jjvdgeer** [comment]
I had the very same idea of using .gitignore for searching and was here to register such a feature request. It seems you beat me to it, but it would be great if this was implemented. Excluding instead of including certain file types would also be a very good feature.

**Comment by alankilborn** [comment]
@leomeyer 

The idea for "exclude binary files" is not new and may be found by searching other open issues.

----------------------------------

@jjvdgeer and @leomeyer 

>.gitignore

Reasonable ideas, indeed.  An alternative would be to set up a Project Panel where the defined project contains on the files that should be searched, and then using the _Find in Projects_ command.

**Comment by jjvdgeer** [comment]
I will have to take a look at how the project stuff works. It wasn't immediately obvious to me.

**Comment by alankilborn** [comment]
>how the project stuff works. It wasn't immediately obvious to me.

Yep, not immediately obvious and really klunky once you get into it.  But, it doesn't get better until people attempt to use it and then put issues in on it's current state.  Please do so if you're inclined.  I just brought it up as an alternative solution that (in theory) should work to keep "non project" files from being searched.

**Comment by klaus101** [comment]
As i'm very interested in this topic, so i'd have a couple of remarks.

- Working via project files might not be appropriate if you search within various, and very highly populated folders.
One can hardly maintain a list of candidates to search here. Eg.: use case: do researches within whole IDE development system folders (eg. Delphi, Lararus etc.), might reach 2 GB.

- The topic itself (ignore binaries) appears to be - maybe? - already acknowledged (assigned to Don Ho), but unfortunately no news over a year.
See: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/9445#issue-comment-box
Are there any news about a possible perspective?

- Preferred: an automatic detection of binary files, or - same good imo!, but probably much more easy to implement: - a simple negative list of file extensions to ignore (see below).

- The advantage of a "negative list of file extensions to ignore" is, imo, that it, although it might be quite long,
not related to diverse programming language file extensions each to maintain. Rather a common thing. With other words: set it once and forget it, more or less. Wherever i might search: search all, but not in those binaries (here: for simpliness identified by file extensions)

- As of today it's already possible to give such a negative list within the "Filter" field, though is is not the natural position for it.
Shortened example (my own filter has 423 chars):
`!*.exe !*.dll !*.dcu !*.zip !*.rar !*.ttf !*.chm !*.pdf !*.xls !*.png !*.ico !*.jpg !*.gif !*.bmp !*.svg !*.tga  // and so on`
More intuitive approach would be:
File exensions to ignore:
`exe,dll,dcu,zip,rar,ttf,chm,pdf,xls,png,ico,jpg,gif,bmp,svg,tga   // and so on`

- To my tests a filter as above does the job quite well and shortens the search time a lot.

However: the search time appears to me to be still too long. At least when searching for a simple text string (not using a regex).
I'm pretty sure it could be very much faster. Simply compare it eg. with grepWin searching within a highly populated folder. Drastic difference! Btw it has an option to include binary files or not, as well as for regex search or text search.
Maybe i'll open a distinc  topic for that.

**Comment by mtrtm** [comment]
Would be great for a user to be able to check a checkbox that says ignore binary and other common files, and have a config file on what extensions and dirs to exclude

**Comment by JesseChisholm** [comment]
Pinging this topic as there doesn't appear to have been any changes in the **Search / Find in Files** dialog, nor in **Settings / Preferences / Searching**.

Nor here: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/9445
